include/material/bingham.tcc

```diff
@@ -0,0 +1,123 @@
+//! Read material properties
+template <unsigned Tdim>
+void mpm::Bingham<Tdim>::properties(const Json& material_properties) {
+  try {
+    density_ = material_properties["density"].template get<double>();
+    youngs_modulus_ =
+        material_properties["youngs_modulus"].template get<double>();
+    poisson_ratio_ =
+        material_properties["poisson_ratio"].template get<double>();
+    tau0_ = material_properties["tau0"].template get<double>();
+    mu_ = material_properties["mu"].template get<double>();
+    critical_shear_rate_ =
+        material_properties["critical_shear_rate"].template get<double>();
+    properties_ = material_properties;
+    status_ = true;
+  } catch (std::exception& except) {
+    console_->error("Material parameter not set: {}\n", except.what());
+  }
+}
+
+//! Elastic tensor is not defined in Bingham model, throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 6> mpm::Bingham<Tdim>::elastic_tensor() {
+
+  throw std::runtime_error("Elastic tensor is not used for this material");
+
+  return Eigen::Matrix<double, 6, 6>::Zero();
+}
+
+//! Compute stress without a particle handle is undefined in the Bingham model,
+//! throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain) {
+
+  throw std::runtime_error(
+      "Stress computation for this material requires a particle handle");
+
+  return Vector6d::Zero();
+}
+
+//! Compute stress
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain,
+    const ParticleBase<Tdim>* ptr) {
+
+  const unsigned phase = 0;
+  auto strain_rate = ptr->strain_rate(phase);
+
+  // Get defintion of D for Bingham, store it as strain_rate
```

Comment:
Definition is spelt incorrectly, maybe say it as `// Convert strain rate to rate of deformation tensor`